### PR TITLE
Vector symbology batch update

### LINKED ISSUE #6421: [FEATURE] New "Merged feature" renderer for polygon and line layers (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#40798
Author: @nyalldawson
QGIS version: 3.18

**[FEATURE] New "Merged feature" renderer for polygon and line layers**

### PR Description:
This renderer merges (or unions/dissolves) the line or polygon features
from a layer prior to rendering them. It's useful for rendering
a polygon layer with overlapping features as one single "coverage" feature,
or a line layer consisting of many smaller component line features
using a regularly spaced marker line or similar.

Internally, this just moves the guts of the existing inverted polygons
renderer to a new base class, as that renderer already had an option
to merge features prior to rendering. Basically it just exposes a new
renderer to users which is the inverted polygon renderer without
the inversion step!

Using single symbol renderer, with marker line symbol:

image

Using merged feature renderer:

image

Fixes #21642
Fixes #27540

### Commits tagged with [need-docs] or [FEATURE]
"[FEATURE] New \"Merged feature\" renderer for polygon and line layers\n\nThis renderer merges (or unions/dissolves) the line or polygon features\nfrom a layer prior to rendering them. It's useful for rendering\na polygon layer with overlapping features as one single \"coverage\" feature,\nor a line layer consisting of many smaller component line features\nusing a regularly spaced marker line or similar.\n\nInternally, this just moves the guts of the existing inverted polygons\nrenderer to a new base class, as that renderer already had an option\nto merge features prior to rendering. Basically it just exposes a new\nrenderer to users which is the inverted polygon renderer without\nthe inversion step!"

### LINKED ISSUE #6571: Expose OGR feature styles via a new "Embedded styling" renderer (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#41957
Author: @nyalldawson
QGIS version: 3.20

**Expose OGR feature styles via a new "Embedded styling" renderer**

### PR Description:
This PR implements the proposal described at https://github.com/qgis/QGIS-Enhancement-Proposals/issues/209, and adds a new "Embedded Symbology" renderer which shows features using their associated symbology from a datasource where features can have "baked in" symbols (such as KML or TAB files).

Currently only supports line symbol conversion, but point and polygon symbol support will come in a follow-up PR.

Peek 2021-03-02 15-22

Tested so far on MapInfo tab files and KML files, but support should extend automatically to all GDAL drivers which implement the OGR feature style API!

Also supports direct conversion of embedded symbols renderers to categorized or rule based renderers, creating categories/rules which define symbology based on the feature IDs:

Peek 2021-03-02 15-23

Sponsored by QGIS Denmark user-group

### Commits tagged with [need-docs] or [FEATURE]

### LINKED ISSUE #6692: interpolated line symbol layer for vector layer (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#42676
Author: @vcloarec
QGIS version: 3.20

**interpolated line symbol layer for vector layer**

### PR Description:
This PR introduces a new ~~renderer~~ symbol layer for line vector layers. This symbol layer is the porting of the mesh layer's edge rendering.

interpolatedLine

The user can associate attributes or expressions to each line string's extremity (or curve). Value is interpolated along the line string between extremities.
The rendering is done with color and/or width 
varying depending on the interpolated value, resulting in a color and width varying line.
The color rendering is set with a color ramp shader widget (like raster and mesh layers).

Funded by Lutra Consulting

Note: text above was changed following the discussion below.

### Commits tagged with [need-docs] or [FEATURE]

### LINKED ISSUE #6715: Vector label legend (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#43121
Author: @mhugent
QGIS version: 3.20

**Vector label legend**

### PR Description:
Sometimes we use layers which are only there for labeling (renderer type 'no symbols'). For such layers, it would be nice to have an entry in the legend showing a placeholder text entry with the labeling font. In other cases we use rule-based labeling and there it is even more important to have legend entries showing the font and the description of the labeling classes. This PR adds a new class QgsVectorLabelLegendNode to show label entries in the legend if 'show label legend' is selected in the legend tab of the vector properties dialog dialog.

It would also be nice to use label entries for context based legend. I'm currently looking if/how this can be added to QgsMapHitTest.

### Commits tagged with [need-docs] or [FEATURE]
"[FEATURE]: Show legend entries for labeling"

### LINKED ISSUE #6790: Expose choice of symbology reference scale for vector layers (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#43826
Author: @nyalldawson
QGIS version: 3.22

**Expose choice of symbology reference scale for vector layers**

### PR Description:
If set, the reference scale indicates the map scale at which symbology and labeling sizes which uses paper-based units
(such as millimeters or points) relate to. The sizes will be scaled accordingly whenever the map is viewed at a different
scale.

For instance, a line layer using a 2mm wide line with a 1:2000 reference scale set will be rendered using 4mm wide lines when
the map is viewed at 1:1000.

Sponsored by North Road, thanks to SLYR

### Commits tagged with [need-docs] or [FEATURE]
"[feature] Expose choice of symbology reference scale for vector\nlayers, in the vector layer properties -> Rendering tab\n\nIf set, the reference scale indicates the map scale at which\nsymbology and labeling sizes which uses paper-based units\n(such as millimeters or points) relate to. The sizes will\nbe scaled accordingly whenever the map is viewed at a different\nscale.\n\nFor instance, a line layer using a 2mm wide line with a 1:2000\nreference scale set will be rendered using 4mm wide lines when\nthe map is viewed at 1:1000.\n\nSponsored by North Road, thanks to SLYR"

### LINKED ISSUE #6948: Expose choice of units for geometry generator symbol layers (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#45125
Author: @nyalldawson
QGIS version: 3.22

**Expose choice of units for geometry generator symbol layers**

### PR Description:
Usually a geometry generator expression must return a geometry in the associated layer's CRS. But this doesn't work in situations where a geometry generator symbol is NOT associated with a layer, e.g. when used in a layout item.

So add a new option to allow users to specify which unit the expression will return geometries in, with choices of map units (the default, previous behavior), millimeters, pixels, inches and points.

~~When millimeters, points, inches or pixels is selected then the @map_geometry variable will be available for the expression,
and contain the feature's geometry in the specified units (relative to the map frame). (The $geometry variable remains available and still in layer CRS, in case an expression needs to calculate the original area, perimeter, etc in real-world units while returning
results in a different unit!)~~

When millimeters, points, inches or pixels is selected then the $geometry variable will be the feature's geometry in the specified units (relative to the map frame).

A step toward fixing #39159, but of interest in its own right !

Here's how it looks:

image

~~I'm interested in feedback regarding the "@map_geometry" variable and better names for this. I'm not happy with @map_geometry, but can't think of an alternative....!~~

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by roya0045**
Fixes #6421 Fixes #6571 Fixes #6692 Fixes #6715 Fixes #6790  Fixes #6948

Proposal to update the vector symbology sections. 

Images still need to be added.

### CODE REVIEW

**Comment by DelazJ** [comment]
@roya0045 I created a dedicated branch for 3.22 so you can open a PR with #6989 changes against master

**Review by DelazJ — COMMENTED** [review]
Doing some review

File: docs/user_manual/style_library/symbol_selector.rst [hunk 1]
```diff
@@ -368,6 +368,11 @@ layer types:
 
      Examples of hashed lines
 
+.. _interpolated_line_symbol:
+
+* **InterPolated Line**: The interpolated line symbol allows to define ways to control
+the changes of witdth and/or color of a line along its trajectory from a set of parameters.
+Those parameter may be feature-specific or static.
```
[the remark this answers is not in the document]
Comment (roya0045) [reply docs/user_manual/style_library/symbol_selector.rst:375]: No worries

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -3137,6 +3183,8 @@ layout legend <layout_legend_item>`. These options include:
   The :guilabel:`Legend placeholder image` helps you :ref:`select an image
   <embedded_file_selector>` for replacement, displayed both in the
   :guilabel:`Layers` panel and the print layout legend.
+* |checkbox| :guilabel:`Sow label legend`: Displays the label entries as their
```
[the remark this answers is not in the document]
Comment (roya0045) [reply docs/user_manual/working_with_vector/vector_properties.rst:3186]: In some instances users may want to use distinct label types to differenciate between elements of a same dataset. For instances road types or water courses or water bodies. Having the text style displayed in the legend would help make clear that distinct label represent different type of entities. It makes sense to have this cartographically.

[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/working_with_vector/vector_properties.rst]: I meant "explain it in the docs" 😄

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -1239,6 +1269,11 @@ You can select an attribute column to use for labeling. By default, the
 :ref:`display field <maptips>` is used. Click |expression| if you want to define
 labels based on expressions - See :ref:`labeling_with_expressions`.
 
+.. note::
+
+   Labels can also be displayed as symbols in the legend, by enabling them in the
+   Legend tab. 
+
```
[the remark this answers is not in the document]
Comment (roya0045) [reply docs/user_manual/working_with_vector/vector_properties.rst:1276]: #6715not sure how to express it otherwise.

**Comment by roya0045** [comment]
@DelazJ The embeded render doesn't have an icon, I'm not sure if I should remove the substitution or not.

**Comment by DelazJ** [comment]
@roya0045 no icon then no substitution possible. We will add one if ever the renderer has an icon (https://github.com/qgis/QGIS/issues/42071)

**Comment by roya0045** [comment]
> @roya0045 no icon then no substitution possible. We will add one if ever the renderer has an icon ([qgis/QGIS#42071](https://github.com/qgis/QGIS/issues/42071))

Good, should be fine now.

**Review by DelazJ — COMMENTED** [review]
@roya0045 Sorry for the delay.

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -3428,6 +3474,8 @@ To do so:
    :width: 1.5em
 .. |mapTips| image:: /static/common/mActionMapTips.png
    :width: 1.5em
+.. |mergedFeaturesSymbol| image:: /static/common/rendererMergedFeatures.svg
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/working_with_vector/vector_properties.rst:3567]: All above is done in https://github.com/qgis/QGIS-Documentation/pull/7368. Only needs to delete the svg file.
```suggestion
.. |mergedFeatures| image:: /static/common/rendererMergedFeatures.png
```

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -836,6 +836,18 @@ From the main dialog, you can:
...
  features.
 
+.. index:: Classes
+   single: Symbology; Merged Features renderer
+.. _merged_renderer:
+
+Merged Features Renderer
+........................
+
+The |mergedFeaturesSymbol| :guilabel:`Merged Features` renderer allows area and line
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/vector_properties.rst:846]:
```suggestion
The |mergedFeatures| :guilabel:`Merged Features` renderer allows area and line
```

File: docs/user_manual/style_library/symbol_selector.rst [hunk 6]
```diff
@@ -441,6 +441,7 @@ layer types:
 
      Examples of lineburst lines
 
+.. _interpolated_line_symbol:
```
Comment (DelazJ) [inline docs/user_manual/style_library/symbol_selector.rst:444]:
```suggestion
```

**Pull request merged**